assets/prometheus-k8s/rules.yaml

```diff
@@ -292,31 +292,28 @@ spec:
       record: cluster:infra_nodes
     - expr: cluster:master_nodes and on(node) cluster:infra_nodes
       labels:
-        label_node_role_kubernetes_io_master_infra: "true"
+        label_node_role_kubernetes_io_infra: "true"
+        label_node_role_kubernetes_io_master: "true"
       record: cluster:master_infra_nodes
-    - expr: kube_node_labels and on(node) (sum(label_replace(node_cpu_info, "node",
-        "$1", "instance", "(.*)")) by (node, package, core) == 2)
-      labels:
-        label_node_hyperthread_enabled: "true"
-      record: cluster:hyperthread_enabled_nodes
-    - expr: sum((cluster:master_nodes * on(node) group_left kube_node_status_capacity_cpu_cores)
-        or on(node) (kube_node_labels * on(node) group_left kube_node_status_capacity_cpu_cores))
-        BY (label_beta_kubernetes_io_instance_type, label_node_role_kubernetes_io)
-      record: cluster:capacity_cpu_cores:sum
-    - expr: sum(((cluster:master_nodes * on(node) group_left(label_node_hyperthread_enabled)
-        cluster:hyperthread_enabled_nodes{label_node_hyperthread_enabled!="true"})
-        or (cluster:master_nodes * on(node) group_left(label_node_hyperthread_enabled)
-        cluster:hyperthread_enabled_nodes{label_node_hyperthread_enabled="true"})
-        * on(node) group_left kube_node_status_capacity_cpu_cores) or on(node) (cluster:hyperthread_enabled_nodes
-        * on(node) group_left kube_node_status_capacity_cpu_cores)  or on(node) ((kube_node_labels
-        unless on(node) cluster:hyperthread_enabled_nodes) * on(node) group_left kube_node_status_capacity_cpu_cores))
-        BY (label_beta_kubernetes_io_instance_type, label_node_hyperthread_enabled,
-        label_node_role_kubernetes_io)
-      record: cluster:capacity_cpu_cores_hyperthread_enabled:sum
+    - expr: cluster:master_infra_nodes or on (node) cluster:master_nodes or on (node)
+        cluster:infra_nodes or on (node) kube_node_labels
+      record: cluster:nodes_roles
     - expr: sum((cluster:master_nodes * on(node) group_left kube_node_status_capacity_memory_bytes)
         or on(node) (kube_node_labels * on(node) group_left kube_node_status_capacity_memory_bytes))
         BY (label_beta_kubernetes_io_instance_type, label_node_role_kubernetes_io)
       record: cluster:capacity_memory_bytes:sum
+    - expr: |
+        sum(cluster:nodes_roles * on (node) group_left() kube_node_status_capacity_cpu_cores)
+        by (label_node_openshift_io_os_id, label_kubernetes_io_arch, label_beta_kubernetes_io_instance_type, label_node_role_kubernetes_io_master, label_node_role_kubernetes_io_infra)
+      record: cluster:capacity_cpus:sum
```

Comment:
What is the distinction between this and cores?  The value reported by Kube node status is subtle and may not represent the real value.  I might call this as `cluster:node_capacity_cpu_cores`.  We should always use `cores` when referring to the units of CPU resources in kube.

Reply:
This one records the number of CPUs as kube sees them.  The number of cores will either be the same as this, or half of this if hyperthreading is enabled.